geth/rpc/client_test.go

```diff
@@ -149,98 +151,93 @@ func (s *RPCTestSuite) TestCallRPC() {
+				func(resultJSON string) {
+					expected := `{"jsonrpc":"2.0","id":67,"result":"5.0"}`
+					s.Equal(expected, resultJSON)
+				},
 			},
-		},
-		{
-			`{"jsonrpc":"2.0","method":"web3_sha3","params":["0x68656c6c6f20776f726c64"],"id":64}`,
-			func(resultJSON string) {
-				expected := `{"jsonrpc":"2.0","id":64,"result":"0x47173285a8d7341e5e972fc677286384f802f8ef42a5ec5f03bbfa254cb01fad"}`
-				s.Equal(expected, resultJSON)
-				progress <- struct{}{}
+			{
+				`{"jsonrpc":"2.0","method":"web3_sha3","params":["0x68656c6c6f20776f726c64"],"id":64}`,
+				func(resultJSON string) {
+					expected := `{"jsonrpc":"2.0","id":64,"result":"0x47173285a8d7341e5e972fc677286384f802f8ef42a5ec5f03bbfa254cb01fad"}`
+					s.Equal(expected, resultJSON)
+				},
 			},
-		},
-		{
-			`{"jsonrpc":"2.0","method":"net_version","params":[],"id":67}`,
-			func(resultJSON string) {
-				expected := `{"jsonrpc":"2.0","id":67,"result":"4"}`
-				s.Equal(expected, resultJSON)
-				progress <- struct{}{}
+			{
+				`{"jsonrpc":"2.0","method":"net_version","params":[],"id":67}`,
+				func(resultJSON string) {
+					expected := `{"jsonrpc":"2.0","id":67,"result":"4"}`
+					s.Equal(expected, resultJSON)
+				},
 			},
-		},
-		{
-			`[{"jsonrpc":"2.0","method":"net_version","params":[],"id":67}]`,
-			func(resultJSON string) {
-				expected := `[{"jsonrpc":"2.0","id":67,"result":"4"}]`
-				s.Equal(expected, resultJSON)
-				progress <- struct{}{}
+			{
+				`[{"jsonrpc":"2.0","method":"net_listening","params":[],"id":67}]`,
+				func(resultJSON string) {
+					expected := `[{"jsonrpc":"2.0","id":67,"result":true}]`
+					s.Equal(expected, resultJSON)
+				},
 			},
-		},
-		{
-			`[{"jsonrpc":"2.0","method":"net_version","params":[],"id":67},{"jsonrpc":"2.0","method":"web3_sha3","params":["0x68656c6c6f20776f726c64"],"id":68}]`,
-			func(resultJSON string) {
-				expected := `[{"jsonrpc":"2.0","id":67,"result":"4"},{"jsonrpc":"2.0","id":68,"result":"0x47173285a8d7341e5e972fc677286384f802f8ef42a5ec5f03bbfa254cb01fad"}]`
-				s.Equal(expected, resultJSON)
-				progress <- struct{}{}
+			{
+				`[{"jsonrpc":"2.0","method":"net_version","params":[],"id":67},{"jsonrpc":"2.0","method":"web3_sha3","params":["0x68656c6c6f20776f726c64"],"id":68}]`,
+				func(resultJSON string) {
+					expected := `[{"jsonrpc":"2.0","id":67,"result":"4"},{"jsonrpc":"2.0","id":68,"result":"0x47173285a8d7341e5e972fc677286384f802f8ef42a5ec5f03bbfa254cb01fad"}]`
+					s.Equal(expected, resultJSON)
+				},
 			},
-		},
-	}
+		}
 
-	cnt := len(rpcCalls) - 1 // send transaction blocks up until complete/discarded/times out
-	for _, r := range rpcCalls {
-		go func(r rpcCall) {
-			resultJSON := rpcClient.CallRaw(r.inputJSON)
-			r.validator(resultJSON)
-		}(r)
-	}
+		var wg sync.WaitGroup
+		for _, r := range rpcCalls {
+			wg.Add(1)
+			go func(r rpcCall) {
+				defer wg.Done()
+				resultJSON := rpcClient.CallRaw(r.inputJSON)
+				r.validator(resultJSON)
```

Comment:
I know that's code existed before PR, but I'm not sure `r.validator` will work as expected when launched from goroutine. Seems like `testify` doesn't handle such cases correctly, but I might be wrong. Anyway, it's not to be fixed here, just to keep in mind that it might the an issue with this test.

Reply:
I am not really sure why you think it won't work as expected. `testify` can handle running assertions in goroutines just fine.

Reply:
@adambabik because I spent yesterday an hour fighting the same problem in other test that needed to do assertions/require in goroutines :) It just doesn't print anything in case of error and it's not easy to spot.

Reply:
I usually run tests so that they fail first and then pass (so as they should be :D). It printed everything nicely and reported failure. Maybe the conditions were a little bit different? Hard to tell but here everything was ok.